src/serializers/type_serializers/union.rs

```diff
@@ -375,38 +300,20 @@ impl TypeSerializer for TaggedUnionSerializer {
         exclude: Option<&Bound<'_, PyAny>>,
         extra: &Extra,
     ) -> Result<S::Ok, S::Error> {
-        let py = value.py();
-        let mut new_extra = extra.clone();
-        new_extra.check = SerCheck::Strict;
-
-        if let Some(tag) = self.get_discriminator_value(value, extra) {
-            let tag_str = tag.to_string();
-            if let Some(&serializer_index) = self.lookup.get(&tag_str) {
-                let selected_serializer = &self.choices[serializer_index];
-
-                match selected_serializer.to_python(value, include, exclude, &new_extra) {
-                    Ok(v) => return infer_serialize(v.bind(py), serializer, None, None, extra),
-                    Err(_) => {
-                        if self.retry_with_lax_check() {
-                            new_extra.check = SerCheck::Lax;
-                            if let Ok(v) = selected_serializer.to_python(value, include, exclude, &new_extra) {
-                                return infer_serialize(v.bind(py), serializer, None, None, extra);
-                            }
-                        }
-                    }
-                }
-            }
-        }
-
-        serde_serialize(
-            value,
-            serializer,
-            include,
-            exclude,
+        match tagged_union_serialize(
+            None,
```

Comment:
@sydney-runkle this is the source of the perf regression; accidentally switched off tagged union serialization optimization in the JSON case.

Reply:
😬 oops. Great find!